### ISSUE TITLE
Unable to remove product attributes

### ISSUE THREAD

**Issue by clintonb-stripe** [stripe/stripe-go#788]
I am working on a [Terraform provider](https://github.com/franckverrot/terraform-provider-stripe/pull/3), and am unable to delete product attributes. My test below demonstrates this.

Is this a bug in the client library, or am is my implementation incorrect (totally feasible given this is my first foray into Go)?

Go version: `go version go1.11.1 darwin/amd64`
Client version: `55.14.0`

```go
import (
	"github.com/stripe/stripe-go"
	"github.com/stripe/stripe-go/client"
	"log"
	"testing"
)

func TestAttributeRemoval(t *testing.T) {
	sc := &client.API{}
	sc.Init("REPLACE-ME", nil)

	tmp := "foo"
	attributes := []*string{&tmp}


	params := stripe.ProductParams{
		Name: stripe.String("Attribute Test"),
		Type: stripe.String("good"),
		Attributes: attributes,
	}

	product, err := sc.Products.New(&params)

	if err != nil {
		t.Errorf("Product creation failed: %v", err)
	}

	params = stripe.ProductParams{
		Attributes: []*string{},
	}
	product, err = sc.Products.Update(product.ID, &params)

	if err != nil {
		t.Errorf("Product update failed: %v", err)
	}

	log.Printf("%v", product)

	if len(product.Attributes) != 0 {
		t.Errorf("Attributes were not deleted: %v", product.Attributes)
	}
}
```

**Comment by brandur-stripe** [comment]
Hi Clinton,

I think you've uncovered a bug in the way that this is "array emptying" is a feature that we never quite got around to implementing in a general purpose way.

A while back we were converting stripe-go's APIs over to use new pointer-based conventions. We got 99% of it, but there were a few places where we left things the old way, one being this feature. I don't think there was any particular reason for that — most likely it was just because so few places actually needed it. `additional_owners` under account is one of the others, but it already had a special workaround called `AdditionalOwnersEmpty`, and we just left that in place.

I think we should be able to get this fixed without too much trouble. I'll take a look in a bit.

**Comment by brandur-stripe** [comment]
Alright, I think I have a fix for this in #789.
